### DISCUSSION TITLE
Fatrop equality constraint problem

### DISCUSSION THREAD

**Discussion by LinusTxtonomy** [casadi/casadi#3738] [Q&A] [answered]
I wanted to try the new fatrop interface with the python direct multiple shooting demo. I enabled automatic structure detection, but that requires the equality option to be set. Since it is multiple shooting I set all elements in the equality vector to True. I then get an error:
```Uncaught Exception: .../casadi/interfaces/fatrop/fatrop_runtime.hpp:217: Structure mismatch: gap-closing constraints must be like this: x_{k+1}-F(xk,uk).```
I am not sure what to do with this error or if I am understanding something wrong about the equality option?
This should be reproducable with the latest nightly build and changing this line: https://github.com/casadi/casadi/blob/72764dc73eab7b935db7e302278508855815883b/docs/examples/python/direct_multiple_shooting.py#L109 to:
```python
solver = nlpsol('solver', 'fatrop', prob, {"structure_detection": "auto", "debug": True, "equality": [True for _ in range(N * x.numel())]})
```

**Comment by jgillis** [comment, marked as answer]
Try flipping the sign of the gap closing constraint..

**Reply by LinusTxtonomy** [reply]
Thank you, that worked. I misunderstood the error message then
